app/models/thread.php

```diff
@@ -0,0 +1,88 @@
+<?php
+class Thread extends AppModel                    
+{
+    public $validation = array('title'=>array('length'=>array('validate_between',1,30),),); 
+
+    public static function getAll($offset, $limit)                
+    {
+        $threads = array();
+        $db = DB::conn();
+        $rows = $db->rows("SELECT * FROM thread LIMIT {$offset}, {$limit}");
+        
+        foreach ($rows as $row) {                    
+            $threads[] = new Thread($row);
+        }
+        
+        return $threads;
+    }   
+
+    public function create(Comment $comment)
+    {
+        $this->validate();
+        $comment->validate();
+        if( $this->hasError() || $comment->hasError()){
+            throw new ValidationException('Invalid Thread or Comment');
+        }
+
+        $db = DB::conn();
+        $db->begin();
+        
+        $db->query('INSERT INTO thread SET title=?, created=NOW()',array($this->title));
+        $this->id = $db->lastInsertId();
+        $this->write($comment);
+
```

Comment:
Why did you use transtractions if you are just dealing with the single table and single query?
and if you use transactions po, dont forget to catch exceptions and rollback the changes